### ISSUE TITLE
Missing 'focusin' and 'focusout' events

### ISSUE THREAD

**Issue by Nightmaster** [ionic-team/stencil#2435]
**Stencil version:**
```
 @stencil/core@1.12.7
```

**I'm submitting a:**
[ x ] bug report
[ ] feature request
[ ] support request

**Current behavior:**
Events [focus in](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusin_event) and [focus out](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusout_event) cannot be declared in TSX

**Expected behavior:**
Possibility to use `onFocusIn` and `onFocusOut` on any HTML element without TypeScript error, and event correctly fired when compiled application is used

**Steps to reproduce:**
Just use code upper and try to compile

**Related code:**

```tsx
// [...]
render = (): void =>
  <div onFocusIn={ console.debug } onFocusOut={ console.debug }>
    // put whatever you want inside
  </div>

// [...]
```

**Other information:**
A pull request to solve the base of the problem is coming. But I did not dig enough to know if it is gonna be enough to fire events on compiled code

**Comment by WickyNilliams** [comment]
Hey it seems that this doesn't actually work.

If i render something like this:

```jsx
<div onFocusIn={e => console.log("focusin", e)}>
  <button>hello world!</button>
</div>
```

I will never see anything logged in the console when the button receives focus.

After inspecting what listeners are registered on the div, i see stencil has added a listener for `focusIn` instead of `focusin`. Event names are case-sensitive so this does not work. Here's a screenshot of the registered listeners in the devtools:

![image](https://user-images.githubusercontent.com/1091390/85412995-20352e00-b562-11ea-93bf-8e417415ef7f.png)


However, if I instead render this:

```jsx
{/* @ts-ignore */}
<div onFocusin={e => console.log("focusin", e)}>
  <button>hello world!</button>
</div>
```

Then I will get a log in the console. But as you can see, it no longer satisfies the TS types. So there seems to be some issue with how stencil is mapping `onFocusIn` in JSX to the actual DOM event `focusin`

**Comment by WickyNilliams** [comment]
Shall i create a new issue for this? I'm happy to attempt a PR if someone could point me to the code that handles mapping JSX to DOM events

**Comment by WickyNilliams** [comment]
eventually created a new issue https://github.com/ionic-team/stencil/issues/2628

**Comment by claviska** [comment]
Fixed in #2745
